vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -40,7 +44,7 @@ def save_inventory(type, parent, hash)
     found = parent.send(type)
     if found.nil?
       found = parent.send("build_#{type}", hash.except(:id))
-      parent.send("#{type}=", found)
+      parent.update_attributes!(type => found)
```

Comment:
> What specific issue does this solve, because I can't create a pattern that fails locally the "old" way?

I think I understand one way the problem could manifest, but I can't see how the current logic allows it.  So, one possibility is has_one-on-a-has_one is not being saved... An example (without the save_inventory)

```ruby
s = ConfiguredSystem.create!
s.create_computer_system! # so computer_system is persisted
s.computer_system.build_operating_system
s.save! # The save will not pass through the intermediate layer because
        # autosave is not set, and the operating system will not be saved.
```

Note that saving deep records does occur if the chain is new all the way down.

```ruby
s = ConfiguredSystem.create!
s.build_computer_system
s.computer_system.build_operating_system
s.save! # Both computer_system and operating_system will be saved
```

What I don't understand is that as we walk the entities, when we get to a has_one and it is found, then we call `update_attributes!` [here](https://github.com/ManageIQ/manageiq/blob/a3b7e9224280ccf9d8691070996e7582a9807ea5/vmdb/app/models/ems_refresh/save_inventory_helper.rb#L45), which is the same as setting the attributes and then calling `save!`, so in the first example above, it should work with `save_inventory` because .update_attributes would get called on the computer_system.

So, what am I missing?